app/main.py

```diff
@@ -1,52 +1,43 @@
-import json
-import xml.etree.ElementTree as ET
-
-
-class Book:
-    def __init__(self, title: str, content: str):
-        self.title = title
-        self.content = content
-
-    def display(self, display_type: str) -> None:
-        if display_type == "console":
-            print(self.content)
-        elif display_type == "reverse":
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown display type: {display_type}")
-
-    def print_book(self, print_type: str) -> None:
-        if print_type == "console":
-            print(f"Printing the book: {self.title}...")
-            print(self.content)
-        elif print_type == "reverse":
-            print(f"Printing the book in reverse: {self.title}...")
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown print type: {print_type}")
-
-    def serialize(self, serialize_type: str) -> str:
-        if serialize_type == "json":
-            return json.dumps({"title": self.title, "content": self.content})
-        elif serialize_type == "xml":
-            root = ET.Element("book")
-            title = ET.SubElement(root, "title")
-            title.text = self.title
-            content = ET.SubElement(root, "content")
-            content.text = self.content
-            return ET.tostring(root, encoding="unicode")
-        else:
-            raise ValueError(f"Unknown serialize type: {serialize_type}")
+from app.book import Book
+from app.utils.handlers import (Display, Print, Serialize,
+                                ConsolePrint, ReversePrint)
+from app.utils.display import ConsoleDisplay, ReverseDisplay
+from app.utils.serializers import JsonSerialize, XmlSerialize
 
 
 def main(book: Book, commands: list[tuple[str, str]]) -> None | str:
     for cmd, method_type in commands:
         if cmd == "display":
-            book.display(method_type)
+            display_handler = get_handler_by_type(method_type, {
+                "console": ConsoleDisplay(),
+                "reverse": ReverseDisplay()
+            })
+            book.display(display_handler)
         elif cmd == "print":
-            book.print_book(method_type)
+            print_handler = get_handler_by_type(method_type, {
+                "console": ConsolePrint(),
+                "reverse": ReversePrint()
+            })
+            book.print_book(print_handler)
         elif cmd == "serialize":
-            return book.serialize(method_type)
+            serialize_handler = get_handler_by_type(method_type, {
+                "json": JsonSerialize(),
+                "xml": XmlSerialize()
+            })
+            return book.serialize(serialize_handler)
+        else:
+            raise ValueError(f"Unknown command: {cmd}")
```

Comment:
Maybe we can also optimize this part due to Open-Closed?
I recommend to create some enum(Command). For example you can use following structure:
```
class Command(Enum):
    DISPLAY = "display"

    @property
    def handlers_dict(self):
        commands_info = {
            Command.DISPLAY: {
                "console": lambda: ConsoleDisplay,
                "reverse": lambda: ReverseDisplay
            }
        }

        return commands_info.get(self, {})

    def get_book_method(self, book: Book):
        methods = {
            Command.DISPLAY: book.display
        }

        return methods.get(self)

    def invoke(self, book: Book, method_type: str):
        method = self.get_book_method(book)
        handler = get_handler_by_type(method_type, self.handlers_dict)

        return method(handler)
```